sycl/doc/extensions/experimental/sycl_ext_oneapi_inter_process_communication.asciidoc

```diff
@@ -0,0 +1,225 @@
+
+== Contact
+
+To report problems with this extension, please open a new issue at:
+
+https://github.com/intel/llvm/issues
+
+
+== Dependencies
+
+This extension is written against the SYCL 2020 revision 10 specification.  All
+references below to the "core SYCL specification" or to section numbers in the
+SYCL specification refer to that revision.
+
+
+== Status
+
+This is an experimental extension specification, intended to provide early
+access to features and gather community feedback.  Interfaces defined in this
+specification are implemented in {dpcpp}, but they are not finalized and may
+change incompatibly in future versions of {dpcpp} without prior notice.
+*Shipping software products should not rely on APIs defined in this
+specification.*
+
+
+== Overview
+
+This extension adds the ability for SYCL programs to share device USM memory
+allocations between processes. This is done by the allocating process creating
+a new IPC memory handle through the new free frunctions and transferring the
+returned handle data to the other processes. The other processes can use the
+handle data to retrieve the corresponding device USM memory.
+
+
+== Specification
+
+=== Feature test macro
+
+This extension provides a feature-test macro as described in the core SYCL
+specification.  An implementation supporting this extension must predefine the
+macro `SYCL_EXT_ONEAPI_INTER_PROCESS_COMMUNICATION` to one of the values defined
+in the table below.  Applications can test for the existence of this macro to
+determine if the implementation supports this feature, or applications can test
+the macro's value to determine which of the extension's features the
+implementation supports.
+
+[%header,cols="1,5"]
+|===
+|Value
+|Description
+
+|1
+|The APIs of this experimental extension are not versioned, so the
+ feature-test macro always has this value.
+|===
+
+=== Extension to `enum class aspect`
+
+[source]
+----
+namespace sycl {
+enum class aspect {
+  ...
+  ext_oneapi_ipc_memory
+}
+}
+----
+
+If a SYCL device has this aspect, that device supports the `get` and `open`
+functions specified in the following section.
+
+
+=== Inter-process communicable memory
+
+
+This extension adds new free functions under the `ipc_memory` experimental
+namespace.
+
+```
+namespace sycl::ext::oneapi::experimental::ipc_memory {
+
+using handle_data_t = std::vector<std::byte>;
+
+handle_data_t get(void *ptr, const sycl::context &ctx);
+
+void put(handle_data_t &handle_data, const sycl::context &ctx);
+
+static void *open(handle_data_t handle_data, const sycl::context &ctx,
+                  const sycl::device &dev);
+
+static void close(void *ptr, const sycl::context &ctx);
+
+}
+```
+
+|====
+a|
+[frame=all,grid=none]
+!====
+a!
+[source]
+----
+handle_data_t get(void *ptr, const sycl::context &ctx)
+----
+!====
+
+_Preconditions:_ `ptr` is a pointer to USM device memory on some device _D_ in
+context `ctx`. `ctx` is the same context as `ptr` was allocated against, using
```

Comment:
```suggestion
_Preconditions:_ `ptr` is a pointer to USM device memory on some device _D_, and
`ctx` is the same context as `ptr` was allocated against, using
```

There is no need to say that _D_ must be in `ctx`.  If we say that `ptr` must have been allocated using `ctx`, then this implies that _D_ is in `ctx` because the allocation routines enforce this.